pkg/server/server.go

```diff
@@ -0,0 +1,59 @@
+package server
+
+import (
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/ipfs/testground/pkg/engine"
+	"github.com/ipfs/testground/pkg/logging"
+	"github.com/pborman/uuid"
+	"go.uber.org/zap"
+)
+
+// _engine is the default engine shared by all commands.
+var (
+	_engine     *engine.Engine
+	_engineErr  error
+	_engineOnce sync.Once
+)
+
+func GetEngine() (*engine.Engine, error) {
+	_engineOnce.Do(func() {
+		_engine, _engineErr = engine.NewDefaultEngine()
+	})
+	return _engine, _engineErr
+}
+
+type Server struct {
```

Comment:
Can we document the HTTP API here? In the future we might want to introduce Swagger for API docs, especially if/as the API becomes richer. But for now, enumerating methods in the following form would suffice to get an API overview:

```
* POST /build: builds ...
```